src/main/cpp/src/cast_string_to_datetime.cu

```diff
@@ -1113,17 +1112,29 @@ std::unique_ptr<cudf::column> parse_timestamp_strings(cudf::strings_column_view
                                                       int64_t default_epoch_day,
                                                       cudf::column_view const& tz_info,
                                                       cudf::table_view const& transitions,
-                                                      bool is_spark_320,
+                                                      int platform_ordinal,
+                                                      int major_version,
+                                                      int minor_version,
+                                                      int fix_version,
                                                       rmm::cuda_stream_view stream,
                                                       rmm::device_async_resource_ref mr)
 {
+  platform_type const platform = spark_rapids_jni::get_platform_type(platform_ordinal);
+  int v                        = major_version * 100 + minor_version * 10 + fix_version;
+  bool is_spark_320            = (platform == platform_type::SPARK && v == 320);
+  bool is_spark_400_or_later   = (platform == platform_type::SPARK && v >= 400);
+  int v_for_db                 = major_version * 100 + minor_version;
+  bool is_db_14_3_or_later     = (platform == platform_type::DATABRICKS && v_for_db >= 1403);
```

Comment:
For better portability and code reuse, let add a custom class in `version.hpp/cpp` for doing all those parsing:
```
struct spark_system {
  spark_system(platform, major, minor, fix) {...}
  platform_type platform() const {...}
  int spark_version() const { return _spark_version; }
private:
 platform_type platform;
 int _spark_version; // can be 320
}
```